hazelcast/src/main/java/com/hazelcast/map/impl/MapSplitBrainHandlerService.java

```diff
@@ -82,9 +81,8 @@ public Runnable prepareMergeRunnable() {
                 }
                 // clear all records either owned or backup
                 recordStore.reset();
+                mapContainer.getIndexes(i).clearIndexes();
```

Comment:
Yeah, here I kind of thought the code is simpler like this, and the overhead is not super significant.